### ISSUE TITLE
Transient test failures

### ISSUE THREAD

**Issue by folksgl** [folksgl/challenger#2]
Running tests do not produce the same results every time. Most times the tests all run to completion, but other times some tests fail. The failing tests are not consistent

**To Reproduce**
Simply compile and run the tests until failures occur

**Expected behavior**
Reproducible test results

**Desktop (please complete the following information):**
 - OS: Ubuntu 18.04 LTS on WSL
 - Version: Master:latest

**Comment by folksgl** [comment]
Error being caused by atoi function in set_piece_positions interpreting characters incorrectly.

**Comment by folksgl** [comment]
Fixed by commit 023d90a9d2a92511427a3d016695136331fd2dc6
Validated by running the following bash script with no failures:
```
#!/bin/bash
for i in {1..10000}

do
    ./runtest > /dev/null
    if [ $? -eq 1 ]; then
        echo FAILED_AT_LEAST_ONCE
        exit 1
    fi
done
```
